test/python/circuit/test_rust_equivalence.py

```diff
@@ -12,17 +12,21 @@
 
 """Rust gate definition tests"""
 
+from functools import partial
 from math import pi
 
 from test import QiskitTestCase
 
 import numpy as np
 
 from qiskit.circuit import QuantumCircuit, CircuitInstruction
+from qiskit.circuit.library.standard_gates import C3XGate, CU1Gate, CZGate, CCZGate
 from qiskit.circuit.library.standard_gates import get_standard_gate_name_mapping
+from qiskit.quantum_info import Operator
 
 SKIP_LIST = {"rx", "ry", "ecr"}
-CUSTOM_MAPPING = {"x", "rz"}
+CUSTOM_NAME_MAPPING = {"mcx": C3XGate()}
+MATRIX_SKIP_LIST = {"c3sx"}
```

Comment:
I thought I saw the rust definition of this matrix using the macro in the code?

Reply:
Yes, the test is being skipped because of the python definition, not the rust one.
`qiskit.circuit.exceptions.CircuitError: "to_matrix not defined for this <class '_SingletonC3SXGate'>"`

Reply:
Hmm, that seems like a bug in the python side then. But lets fix that in a separate PR

Reply:
I will keep it in mind!

Reply:
#12742